squad/core/management/commands/update_project_statuses.py

```diff
@@ -32,19 +33,39 @@ def add_arguments(self, parser):
             type=valid_date,
             help="Start date for project status updates (default: 6 months before current date, format: YYYY-MM-DD)."
         )
+        parser.add_argument(
+            '--date-end',
+            dest="date_end",
+            default=datetime.now(),
+            type=valid_date,
+            help="End date for project status updates (default: today, format: YYYY-MM-DD)."
+        )
 
     def handle(self, *args, **options):
         self.options = options
 
-        builds = Build.objects.filter(
-            datetime__gt=timezone.make_aware(self.options['date_start']),
+        build_count = Build.objects.filter(
+            datetime__range=(timezone.make_aware(self.options['date_start']),
+                             timezone.make_aware(self.options['date_end'])),
             status__finished=True
-        )
-        total = builds.count()
-        for index, build in enumerate(builds):
-            ProjectStatus.create_or_update(build)
-
-            if index % 100 == 0:
-                logger.info('Progress: {1:>2}%[{0:10}]'.format(
-                    '#' * int((index + 1) * 10 / total),
-                    int((index + 1) * 100 / total)), end='')
+        ).count()
+
+        print("Updating ProjectStatus objects from builds...")
+        print("Total build count in selected date range: %s" % build_count)
+
+        offset = 0
+        step = 50
+        while offset <= build_count:
+            builds = Build.objects.filter(
+                datetime__range=(
+                    timezone.make_aware(self.options['date_start']),
+                    timezone.make_aware(self.options['date_end'])),
+                status__finished=True
+            )[offset:offset + step]
+            for index, build in enumerate(builds):
+                ProjectStatus.create_or_update(build)
+                sys.stdout.write('\rProgress: {0}/{1}'.format(
+                    offset + index + 1, build_count))
+                sys.stdout.flush()
```

Comment:
please don't use sys.stdout as we don't have a way of turning  that off when running tests; use logger instead

Reply:
also, logging at every single object is too much; think of several hundred thousand objects ... and sending \r will usually make logs unreadable.